Software/Development/SemesterProject/App/closedLoopControlManager.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * closedLoopControlManager.c
+ *
+ *  Created on: 25. apr. 2019
+ *      Author: Nicolai Fransen
+ */
+
+#include "closedLoopControlManager.h"
+
+void runClosedLoopControl(void)
+{
+    float movementReference = 0;
+    float iqReference = 0, idReference = 0;
+    float abcCurrents[2];
+    float theta = 0;
+    volatile dqObject dqCurrents, dqVoltages;
+    alphaBetaObject abVoltages;
+
+    movementReference = getMovementReference();         // Reads torque or speed reference
+
+    iqReference = getIqReference(movementReference);    // Calculate iq reference from movement reference
+    idReference = calculateIdReference();               // Calculate id reference
+
+    getCurrentMeasurements(&abcCurrents[0]);            // Reads current measurements
+    theta = readRotorElecAngleRad();                    // reads electrical angle in radians
+
+    dqCurrents = abc2dq(&abcCurrents[0], theta);        // Transform current measurements from abc->dq
+
+    // Calculate voltage references from current PI-controllers
+    dqVoltages.qComponent = PICalculationIQ(iqReference, dqCurrents.qComponent);

```

Comment:
this could be hidden by sending dqObject (one additional abstraction interface)

Reply:
So now i have created an function which takes an array of current references and the dqObject, and returns the voltage references in a dqObject.